### PR TITLE
GUA-435: Deploy stack with the secure pipeline

### PR DESCRIPTION
**Pull request by alex9smith**
Configure this repository to deploy the backend to the build AWS account using the dev platform secure pipeline. 

Mainly following [the documentation here](https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3059908609/Deploy+a+SAM+application+with+secure+pipelines), plus any tweaks needed to our SAM template.

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -61,15 +79,23 @@ Resources:
           Value: !Ref 'Environment'
         - Key: Source
           Value: "https://github.com/alphagov/di-account-management-backend/blob/main/infrastructre/template.yaml"
+      PointInTimeRecoverySpecification:
+              PointInTimeRecoveryEnabled: true
```
Comment (huwd) [inline infrastructure/template.yaml:83]: Can I just double check the indentation is right on this one @divyaparameswaran ?

I know YAML can be a bit of a pain about indents, this looks like it's a few below line 82?

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -61,15 +79,23 @@ Resources:
...
   # TxMA Event Queue (Dummy for now)
   ######################################
   TxMAInputDummyQueue:
     Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !GetAtt QueueKmsKey.Arn
     DeletionPolicy: Delete
     UpdateReplacePolicy: Delete
-
+
```
Comment (huwd) [inline infrastructure/template.yaml:98]: Ok really non blocking but might want to clear out that whitespace there too.

I think Alex and I both have trim whitespace on here. I think if I saved the file it'd auto edit this line too, which might be a bit odd for any future reviewers.

Here's where I've got it turned on:
image

File: infrastructure/template.yaml [hunk 3]
```diff
@@ -232,17 +268,24 @@ Resources:
     Properties:
       LogGroupName: !Sub "${AWS::StackName}-format-user-service-log-group"
       RetentionInDays: 30
+      KmsKeyId: !GetAtt LoggingKmsKey.Arn
+
```
Comment (huwd) [inline infrastructure/template.yaml:272]: Total nitpick... I feel like i've become the whitespace police 🤦🏻 

```suggestion
```

**Review by huwd — COMMENTED** [review]
Looks good, can I just check on that indention comment?

Otherwise your'e welcome to take or leave my pedantic whitespace comments.

All the config looks great :)

**Comment by divyaparameswaran** [comment]
I have updated the indentation and the VS code config . If it is still not correct I will take a look again .

**Review by huwd — APPROVED** [review]

**Pull request merged**